src/ProtocolStreamReader.php

```diff
@@ -30,33 +30,32 @@ public function __construct($input)
     {
         $this->input = $input;
         Loop\addReadStream($this->input, function() {
-            while (($c = fgetc($this->input)) !== false) {
-                $this->buffer .= $c;
-                switch ($this->parsingMode) {
-                    case ParsingMode::HEADERS:
-                        if ($this->buffer === "\r\n") {
-                            $this->parsingMode = ParsingMode::BODY;
-                            $this->contentLength = (int)$this->headers['Content-Length'];
-                            $this->buffer = '';
-                        } else if (substr($this->buffer, -2) === "\r\n") {
-                            $parts = explode(':', $this->buffer);
-                            $this->headers[$parts[0]] = trim($parts[1]);
-                            $this->buffer = '';
+            $c = fgetc($this->input);
```

Comment:
You are right, it will wait for EOF, I thought it would check if the buffer still has data. Would it help if we instead used `fgets`? `stream_select` is used by `Loop\addReadStream`, but we need some way to only read until the buffer is empty... If it always reads until EOF we have a problem

Is the problem really a Windows limitation, not a PHP bug? Because how do all the other language server like the TS server do it then? They all talk through STDIO.